packages/metadata/src/types.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/metadata
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT'
+
+/**
+ * Decorator function types
+ */
+export type DecoratorType =
+  | ClassDecorator
+  | PropertyDecorator
+  | MethodDecorator
+  | ParameterDecorator;
+
+/**
+ * A strongly-typed metadata accessor via reflection
+ * @typeparam T Type of the metadata value
+ * @typeparam D Type of the decorator
+ */
+export class MetadataAccessor<T, D extends DecoratorType = DecoratorType> {
+  private constructor(public readonly key: string) {}
+
+  toString() {
```

Comment:
Isn't it cleaner and easier to access the `.key` property directly?

Reply:
Please note toString() works for both `MetadataAccessor` and `string` so that we don't to test if the key is a string or not.